pkg/proxy/proxy.go

```diff
@@ -417,9 +417,10 @@ func (s *Proxy) addCluster(obj interface{}) {
 		}
 	}
 
+	klog.V(4).Infof("mcService.Spec.ClusterIP '%s', service.Spec.ClusterIP '%s'", mcService.Spec.ClusterIP, service.Spec.ClusterIP)
 	// populates the kubernetes apiEndpoint and kubesphere apiEndpoint
-	cluster.Spec.Connection.KubernetesAPIEndpoint = fmt.Sprintf("https://%s:%d", mcService.Spec.ClusterIP, kubernetesPort)
-	cluster.Spec.Connection.KubeSphereAPIEndpoint = fmt.Sprintf("http://%s:%d", mcService.Spec.ClusterIP, kubespherePort)
+	cluster.Spec.Connection.KubernetesAPIEndpoint = fmt.Sprintf("https://%s:%d", service.Spec.ClusterIP, kubernetesPort)
+	cluster.Spec.Connection.KubeSphereAPIEndpoint = fmt.Sprintf("http://%s:%d", service.Spec.ClusterIP, kubespherePort)
```

Comment:
When a cluster is added the first time, we create an `svc` for the cluster and the `mcService.spec.clusterIp` is the same as the `svc`. I think using the `mcService` is safe.

Reply:
It is quite strange, let's check https://github.com/kubernetes/client-go/blob/v0.18.6/kubernetes/typed/core/v1/service.go#L112
As it shows,the passed in "mcService" is not updated. Only the "result" shows updated service object.
Besides that, log also shows "mcService" is empty when cluster added the first time

Reply:
Nice catch! Thanks for your contribution.